pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -837,7 +837,7 @@ func (p *staticPolicy) updatePodReusableMemory(pod *v1.Pod, container *v1.Contai
 	// If pod entries to m.initContainersReusableMemory other than the current pod exist, delete them.
 	for uid := range p.initContainersReusableMemory {
 		if podUID != uid {
-			delete(p.initContainersReusableMemory, podUID)
+			delete(p.initContainersReusableMemory, uid)
```

Comment:
/ok-to-test
Compared with the implementation of cpuManager, it should be Fixed here

Reply:
Like you can see it is using the different variable name under the loop
https://github.com/kubernetes/kubernetes/blob/95390e6476fca15b0f55cde2c142ec18ca57aee0/pkg/kubelet/cm/cpumanager/policy_static.go#L227
So the behavior here is ok, we want to delete pod from re-usable pods once we are working on the pod with different UID.

Reply:
You are right, but the current code alway deletes the same current podUID inner the `for `iterator. For example, while admitting a pod with 2 init-containers and 2 app-containers. Since caculating memory for the 4 containers, every time in `updatePodReusableMemory(pod *v1.Pod, container *v1.Container, memoryBlocks []state.Block)` , the current code always clear the p.initContainersReusableMemory[podUID] for each container caculating of the current one pod, it's not correct.

Reply:
> Like you can see it is using the different variable name under the loop
> https://github.com/kubernetes/kubernetes/blob/95390e6476fca15b0f55cde2c142ec18ca57aee0/pkg/kubelet/cm/cpumanager/policy_static.go#L227
> 
> 
> So the behavior here is ok, we want to delete pod from re-usable pods once we are working on the pod with different UID.
```

	// If pod entries to m.initContainersReusableMemory other than the current pod exist, delete them.
	for uid := range p.initContainersReusableMemory {
		if podUID != uid {
			delete(p.initContainersReusableMemory, podUID)  // ************while we are working on a new pod with different UID, we should delete other pod from re-usable, not the current new one
```

Reply:
So we have a pod with 4 containers, once you create it, the memory manager sequentially runs `Allocate` for each container, but it calls the allocate with the same pod, so it will not delete the pod from the `initContainersReusableMemory`, why do you think it should
```
every time in updatePodReusableMemory(pod *v1.Pod, container *v1.Container, memoryBlocks []state.Block) , the current code always clear the p.initContainersReusableMemory[podUID] for each container caculating of the current one pod, it's not correct.```

Reply:
ok I see now